client/src/components/Sidebar/ChatContent.js

```diff
@@ -18,14 +19,20 @@ const useStyles = makeStyles((theme) => ({
     color: "#9CADC8",
     letterSpacing: -0.17,
   },
+  unreadCounter: {
+    alignItems: "flex-end",
+    padding: "10px",
+  },
 }));
 
 const ChatContent = (props) => {
   const classes = useStyles();
 
   const { conversation } = props;
   const { latestMessageText, otherUser } = conversation;
-
+  // finds the length of unread messages sent by the other user
+  const unreadMessagesCount = conversation.messages.filter(message => message.read === false 
+    && message.senderId === otherUser.id).length;
```

Comment:
This line counts unread messages on every render. I think it would be more efficient if this logic is placed inside Redux reducer (of conversations).

Reply:
done